pkg/apis/resource/types.go

```diff
@@ -157,12 +157,21 @@ type ResourceSliceSpec struct {
 	//
 	// The names of the SharedCounters must be unique in the ResourceSlice.
 	//
-	// The maximum number of counters in all sets is 32.
+	// The maximum number of counter sets in a ResourceSlice is 32 and the
+	// total number of counters across counter sets and counter set mixins
+	// in a single ResourceSlice must not exceed 256.
```

Comment:
For visibility as that thread is now on an outdated line of code: https://github.com/kubernetes/kubernetes/pull/131357#discussion_r2210695096

We are trying hard to ensure via validation rules that DRA driver authors get obvious errors when they submit a slice which is too large to be stored, while at the same time not limiting individual slices to unusably small sizes. I think we are doing a reasonably good job at this (for example, actually trying to create large slices in integration tests), but I am not able to convince myself that our limits achieve that goal under all circumstances.

IMHO this is good enough for ResourceSlices, which after all are created by trusted system components.